sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -62,6 +62,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] {
   /** Returns a Seq of the children of this node */
   def children: Seq[BaseType]
 
+  lazy val childrenSet:Set[TreeNode[_]] = children.toSet
```

Comment:
how about naming it `containsChild` so that we can use it like `containsChild(arg)`?